plugins/org/src/components/Cards/OwnershipCard/useGetEntities.ts

```diff
@@ -57,78 +58,122 @@ const getQueryParams = (
   return qs.stringify({ filters }, { arrayFormat: 'repeat' });
 };
 
-const getOwnersEntityRef = (owner: Entity): string[] => {
-  let owners = [stringifyEntityRef(owner)];
-  if (owner.kind === 'User') {
-    const ownerGroups = getEntityRelations(owner, RELATION_MEMBER_OF, {
-      kind: 'Group',
-    });
-    const ownerGroupsName = ownerGroups.map(ownerGroup =>
-      stringifyEntityRef({
-        kind: ownerGroup.kind,
-        namespace: ownerGroup.namespace,
-        name: ownerGroup.name,
-      }),
+const getMemberOfEntityRefs = (owner: Entity): string[] => {
+  const parentGroups = getEntityRelations(owner, RELATION_MEMBER_OF, {
+    kind: 'Group',
+  });
+
+  const ownerGroupsNames = parentGroups.map(({ kind, namespace, name }) =>
+    stringifyEntityRef({
+      kind,
+      namespace,
+      name,
+    }),
+  );
+
+  const {
+    kind,
+    metadata: { namespace, name },
+  } = owner;
+
+  return [...ownerGroupsNames, stringifyEntityRef({ kind, namespace, name })];
+};
+
+const isEntity = (entity: Entity | undefined): entity is Entity =>
+  entity !== undefined;
+
+const getChildOwnershipEntityRefs = async (
+  entity: Entity,
+  catalogApi: CatalogApi,
+): Promise<string[]> => {
+  const childGroups = getEntityRelations(entity, RELATION_PARENT_OF, {
+    kind: 'Group',
+  });
+
+  const hasChildGroups = childGroups.length > 0;
+
+  if (hasChildGroups) {
+    const entityRefs = childGroups.map(
+      ({ kind, namespace, name }) => `${kind}:${namespace}/${name}`,
     );
-    owners = [...owners, ...ownerGroupsName];
+    const childGroupResponse = await catalogApi.getEntitiesByRefs({
+      fields: ['kind', 'metadata.namespace', 'metadata.name'],
+      entityRefs,
+    });
+    const childGroupEntities = childGroupResponse.items.filter(isEntity);
+
+    return (
+      await Promise.all(
+        childGroupEntities.map(childGroupEntity =>
+          limiter(() =>
+            getChildOwnershipEntityRefs(childGroupEntity, catalogApi),
+          ),
+        ),
+      )
+    ).flatMap(aggregated => aggregated);
```

Comment:
might be worth deduplicating these btw (probably all the way up at the original call site, of course, since it's recursive)

Reply:
Let's merge for now and you can have a mental note of improving that in the future perhaps?

Reply:
Sounds good to me, thanks for highlighting it!